### PR TITLE
Ghosts are now able to see N2O with the analyze air verb. 

### PR DESCRIPTION
**Pull request by 1080pCat**
## What Does This PR Do

Lets ghosts scan for and see n2o with their analyze air verb.

Edit:
Since slightly refactored/standardized the code chunk.
## Why It's Good For The Game

It was a oversight with #20891, and the normal hand held one can, so ghosts should be able to, to.

## Testing

Checked if n2o showed up, it did, checked if agent b was still unknown, it was. 
## Changelog
:cl:
fix: Fixed a oversight preventing ghosts from seeing N2O with their analyze air verb.
/:cl:

### CODE REVIEW

**Review by PidgeyThePirate — APPROVED** [review]
Follows the template laid out, don't see any issues elsewhere.

File: code/modules/mob/dead/observer/observer_base.dm [hunk 1]
```diff
@@ -552,8 +552,9 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		var/n2_concentration = environment.nitrogen/total_moles
 		var/co2_concentration = environment.carbon_dioxide/total_moles
 		var/plasma_concentration = environment.toxins/total_moles
+		var/n2o_concentration = environment.sleeping_agent/total_moles
```
[the remark this answers is not in the document]
Comment (1080pCat) [reply code/modules/mob/dead/observer/observer_base.dm]: Done

File: code/modules/mob/dead/observer/observer_base.dm [hunk 2]
```diff
@@ -572,6 +573,9 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		if(plasma_concentration > 0.01)
 			to_chat(src, "<span class='warning'>Plasma: [round(plasma_concentration*100)]% ([round(environment.toxins,0.01)] moles)</span>")
 
+		if(n2o_concentration > 0.01)
+			to_chat(src, "<span class='warning'>N2O: [round(n2o_concentration*100)]% ([round(environment.sleeping_agent,0.01)] moles)</span>")
```
[the remark this answers is not in the document]
Comment (1080pCat) [reply code/modules/mob/dead/observer/observer_base.dm]: Done

File: code/modules/mob/dead/observer/observer_base.dm [hunk 3]
```diff
@@ -552,8 +552,9 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		var/n2_concentration = environment.nitrogen/total_moles
 		var/co2_concentration = environment.carbon_dioxide/total_moles
 		var/plasma_concentration = environment.toxins/total_moles
+		var/n2o_concentration = environment.sleeping_agent/total_moles
 
-		var/unknown_concentration =  1-(o2_concentration+n2_concentration+co2_concentration+plasma_concentration)
+		var/unknown_concentration =  1-(o2_concentration+n2_concentration+co2_concentration+plasma_concentration+n2o_concentration)
```
[the remark this answers is not in the document]
Comment (1080pCat) [reply code/modules/mob/dead/observer/observer_base.dm]: Done

File: code/modules/mob/dead/observer/observer_base.dm [hunk 4]
```diff
@@ -544,39 +544,43 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
+		if(n2o_concentration > 0.01)
+			to_chat(src, "<span class='warning'>N2O: [round(n2o_concentration * 100)]% ([round(environment.sleeping_agent, 0.01)] moles)</span>")
 
 		if(unknown_concentration > 0.01)
-			to_chat(src, "<span class='warning'>Unknown: [round(unknown_concentration*100)]% ([round(unknown_concentration*total_moles,0.01)] moles)</span>")
+			to_chat(src, "<span class='warning'>Unknown: [round(unknown_concentration * 100)]% ([round(unknown_concentration*  total_moles, 0.01)] moles)</span>")
 
-		to_chat(src, "<span class='notice'>Temperature: [round(environment.temperature-T0C,0.1)]&deg;C</span>")
-		to_chat(src, "<span class='notice'>Heat Capacity: [round(environment.heat_capacity(),0.1)]</span>")
+		to_chat(src, "<span class='notice'>Temperature: [round(environment.temperature-T0C, 0.1)]&deg;C</span>")
```
Comment (Burzah) [inline code/modules/mob/dead/observer/observer_base.dm:582]:
```suggestion
		to_chat(src, "<span class='notice'>Temperature: [round(environment.temperature - T0C, 0.1)]&deg;C</span>")
```
One more right here. All the other changes look good. Thank you for doing that.

File: code/modules/mob/dead/observer/observer_base.dm [hunk 5]
```diff
@@ -544,39 +544,43 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
-		var/co2_concentration = environment.carbon_dioxide/total_moles
-		var/plasma_concentration = environment.toxins/total_moles
+		var/o2_concentration = environment.oxygen / total_moles
+		var/n2_concentration = environment.nitrogen / total_moles
+		var/co2_concentration = environment.carbon_dioxide / total_moles
+		var/plasma_concentration = environment.toxins / total_moles
+		var/n2o_concentration = environment.sleeping_agent / total_moles
 
-		var/unknown_concentration =  1-(o2_concentration+n2_concentration+co2_concentration+plasma_concentration)
+		var/unknown_concentration =  1-(o2_concentration + n2_concentration + co2_concentration + plasma_concentration + n2o_concentration)
```
Comment (Burzah) [inline code/modules/mob/dead/observer/observer_base.dm:557]:
```suggestion
		var/unknown_concentration =  1 - (o2_concentration + n2_concentration + co2_concentration + plasma_concentration + n2o_concentration)
```
This one too. Adding spaces for these operators makes the code easier to read.

**Review by lewcc — COMMENTED** [review]
Code looks good, just address burzah's change.

File: code/modules/mob/dead/observer/observer_base.dm [hunk 6]
```diff
@@ -544,39 +544,43 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
-		var/co2_concentration = environment.carbon_dioxide/total_moles
-		var/plasma_concentration = environment.toxins/total_moles
+		var/o2_concentration = environment.oxygen / total_moles
+		var/n2_concentration = environment.nitrogen / total_moles
+		var/co2_concentration = environment.carbon_dioxide / total_moles
+		var/plasma_concentration = environment.toxins / total_moles
+		var/n2o_concentration = environment.sleeping_agent / total_moles
 
-		var/unknown_concentration =  1-(o2_concentration+n2_concentration+co2_concentration+plasma_concentration)
+		var/unknown_concentration =  1 - (o2_concentration + n2_concentration + co2_concentration + plasma_concentration + n2o_concentration)
```
Comment (Burzah) [inline code/modules/mob/dead/observer/observer_base.dm:557]:
```suggestion
		var/unknown_concentration = 1 - (o2_concentration + n2_concentration + co2_concentration + plasma_concentration + n2o_concentration)
```
One extra space there, that's the last thing.

**Review by Burzah — APPROVED** [review]

**Review by Sirryan2002 — APPROVED** [review]

**Review by lewcc — APPROVED** [review]

**Pull request merged**